pilot/pkg/model/push_context.go

```diff
@@ -763,6 +779,50 @@ func (ps *PushContext) initServiceAccounts(env *Environment, services []*Service
 	}
 }
 
+// Caches list of authentication policies
+func (ps *PushContext) initAuthNPolicies(env *Environment) error {
+	authNPolicies, err := env.List(schemas.AuthenticationPolicy.Type, NamespaceAll)
```

Comment:
You also need to check mesh policy config, `schemas.AuthenticationMeshPolicy.Type`

Reply:
I spent far too much time trying to figure out why things were broken.  ^^ I completely missed that the existing configstore was doing two different lookups: `AuthenticationMeshPolicy` and `AuthenticationPolicy`, sigh.  Thanks @diemtvu for saving my sanity